### PR TITLE
[browser] Add sandboxing. JB#51740

### PR DESCRIPTION
**Pull request by rainemak**

### CODE REVIEW

File: oneshot.d/browser-cleanup-startup-cache [hunk 1]
```diff
@@ -3,5 +3,5 @@
 rm -rf ~/.mozilla/mozembed/startupCache
 rm -rf ~/.mozilla/captiveportal/startupCache
 # New startup cache location
-rm -rf ~/.local/share/sailfish-browser/.mozilla/startupCache
```
Comment (Tomin1) [inline oneshot.d/browser-cleanup-startup-cache:6]: Are these paths actually correct? I would think that there needs to be `org.sailfishos` directory before `browser`.

File: oneshot.d/browser-cleanup-startup-cache [hunk 2]
```diff
@@ -3,5 +3,5 @@
...
-rm -rf ~/.local/share/sailfish-browser-captiveportal/.mozilla/startupCache
+rm -rf ~/.local/share/browser/.mozilla/startupCache
```
Comment (Tomin1) [inline oneshot.d/browser-cleanup-startup-cache:6]: Are these paths actually correct? I would think that there needs to be org.sailfishos directory before browser.

File: sailfish-browser-captiveportal.desktop [hunk 3]
```diff
@@ -5,5 +5,9 @@ Name=Captive Portal Browser
...
 X-MeeGo-Translation-Catalog=sailfish-browser-captiveportal
 Icon=icon-launcher-browser
-Exec=invoker -s --type=browser -G /usr/bin/sailfish-browser-captiveportal -captiveportal %U
-Comment=Sailfish UI application
+Exec=/usr/bin/sailjail -p sailfish-browser-captiveportal.desktop /usr/bin/sailfish-browser-captiveportal -captiveportal %U
+Comment=Sailfish captive portal
+[X-Sailjail]
+Permissions=CaptivePortal;Privileged;Audio;Location;Internet;Downloads
+OrganizationName=org.sailfishos
+ApplicationName=browser.captiveportal
```
Comment (Tomin1) [inline sailfish-browser-captiveportal.desktop:13]: Originally I envisioned that ApplicationName can't have a dot. I hoped dashes and underscores would suffice. Well, if it works...

Comment (rainemak) [reply sailfish-browser-captiveportal.desktop:13]: Me too.

File: settings/browser.qml [hunk 4]
```diff
@@ -12,24 +11,18 @@
 
 import QtQuick 2.0
 import Sailfish.Silica 1.0
+import org.sailfishos.browser.settings 1.0
 import org.nemomobile.dbus 2.0
+import com.jolla.settings 1.0
 
-Page {
+ApplicationSettings {
     id: page
```
Comment (Tomin1) [inline settings/browser.qml:19]: No need to keep this id.

File: oneshot.d/browser-cleanup-startup-cache (hunk 2 above)
Comment (rainemak) [reply oneshot.d/browser-cleanup-startup-cache:6]: Well spotted!

File: sailfish-browser-captiveportal.desktop (hunk 3 above)
Comment (rainemak) [reply sailfish-browser-captiveportal.desktop:13]: why didn't I just have "captiveportal" there's no need for this "browser.". You're right.

**Review by Tomin1 — APPROVED** [review]
LGTM

**Pull request merged**